cjwkernel/types.py

```diff
@@ -592,26 +653,22 @@ class I18nMessage:
     args: Dict[str, Union[int, float, str]] = field(default_factory=dict)
     """Arguments (empty if message does not need any -- which is common)."""
 
-    source: Dict[str, str] = field(default_factory=dict)
-    """An indication of where the message is coming from.
-        - An empty dict means it's coming from workbench itself
-        - A dict with key `module_id` means it's coming from a module
-        - A dict with key `library` indicates it's coming from some of our supported libraries (e.g. `cjwmodule`)
-    """
+    source: Optional[I18nMessageSource] = None
+    """An indication of where the message is coming from. `None` means it's coming from workbench itself."""
```

Comment:
`"""Where the message comes from, or `None` if it comes from Workbench proper."""`

(trying to fit 80 char line limit)